src/panels/lovelace/cards/hui-thermostat-card.ts

```diff
@@ -0,0 +1,309 @@
+  }
+
+  public setConfig(config: Config) {
+    if (!config.entity || config.entity.split(".")[0] !== "climate") {
+      throw new Error("Specify an entity from within the climate domain.");
+    }
+
+    this.config = config;
+  }
+
+  protected render() {
+    const stateObj = this.hass!.states[this.config!.entity];
+    const sizeClass = this.clientWidth > 450 ? "large" : "small"; // Use Class Map
+    return html`
+      ${this.renderStyle()}
+      <ha-card
+        class="${stateObj.attributes.operation_mode} ${sizeClass}">
+        <div id="root">
+          <div
+            .hass="${this.hass}"
+            .config="${this.config}"
+            .localize="${this.localize}"
+            id="thermostat"
+          ></div>
+          <div id="tooltip">
+            <div class="title">Upstairs</div>
+            <div class="current-temperature">
+              <span class="current-temperature-text">${
+                stateObj.attributes.current_temperature
+              }</span><span class="uom">&deg;F</span>
+            </div>
+            <div class="climate-info">
+            <div id="set-temperature">${stateObj.attributes.temperature}</div>
+            <div class="current-mode">${this.localize(
+              `state.climate.${stateObj.state}`
+            )}</div>
+            <div class="modes">
+              ${stateObj.attributes.operation_list.map((modeItem) =>
+                this._renderIcon(modeItem, stateObj.attributes.operation_mode)
+              )}
+            </div>
+          </div>
+        </div>
+      </ha-card>
+    `;
+  }
+
+  protected shouldUpdate(changedProps) {
+    if (changedProps.get("hass")) {
+      return changedProps.get("hass").states[this.config!.entity] !==
+        this.hass!.states[this.config!.entity]
+        ? true
+        : false;
+    }
+    return changedProps;
+  }
+
+  protected firstUpdated(changedProps) {
+    const stateObj = this.hass!.states[this.config!.entity];
+
+    $("#thermostat", this.shadowRoot).roundSlider({
+      radius: this.clientWidth / 3,
+      min: stateObj.attributes.min_temp || thermostatConfig.min,
+      max: stateObj.attributes.max_temp || thermostatConfig.max,
+      step: thermostatConfig.step,
+      circleShape: thermostatConfig.circleShape,
+      sliderType: thermostatConfig.sliderType,
+      startAngle: thermostatConfig.startAngle,
+      width: thermostatConfig.width,
+      handleSize: thermostatConfig.handleSize,
+      lineCap: thermostatConfig.lineCap,
+      showTooltip: thermostatConfig.showTooltip,
+      change: (value) => this._setTemperature(value),
+      drag: (value) => this._dragEvent(value),
+    });
+    this.roundSlider = $("#thermostat", this.shadowRoot).data("roundSlider");
+  }
+
+  protected updated(changedProps) {
+    const stateObj = this.hass!.states[this.config!.entity];
+
+    this.roundSlider.setValue(stateObj.attributes.temperature);
+    this.shadowRoot!.querySelector(".current-mode")!.innerHTML = this.localize(
```

Comment:
why is this needed? It seems like it is rendered correctly already in the template?

Reply:
This was my attempt to get it to change when the mode changed.It was late and I forgot to delete it. When you click the mode the current mode text doesnt change and the selected icon doesnt change

Reply:
why not? do we need a temporary override while the backend processes things?

Reply:
(if so, that would be for another PR…)